### PR TITLE
feat(discover2): Revised navigation flow for Discover

### PR DESCRIPTION
**Pull request by dashed**
Revise the Discover navigation flow to enforce the drilldown.

Kapture 2020-06-11 at 22 12 56

Other changes include:

- On the Discover results page, an `id` field is added to the eventview object prior to generating the API payload only if the eventview has no aggregates and contains no `id` field. 

- `count(column)` aggregates no longer link to the drilldown CTA. However, `count_unique(column)` will still have the drilldown CTA.

Screen Shot 2020-06-11 at 10 15 09 PM

Screen Shot 2020-06-11 at 10 09 38 PM

## TODO

- [x] update acceptance tests
- [x] fix styling for preprend columns

### CODE REVIEW

File: src/sentry/static/sentry/app/views/eventsV2/table/index.tsx [hunk 1]
```diff
@@ -75,11 +75,23 @@ class Table extends React.PureComponent<TableProps, TableState> {
...
+
+    const hasAggregates = eventView.getAggregateFields().length > 0;
+    const hasEventIdColumn = eventView.getFields().includes('id');
+
+    if (!hasAggregates && !hasEventIdColumn) {
+      eventView = eventView.withNewColumn({
+        kind: 'field',
+        field: 'id',
+      });
+    }
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/table/index.tsx:93]: The API endpoint will automatically append the event id when a result set has no aggregates, as we needed it in the past to create links. You shouldn't need this.

File: src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx [hunk 2]
```diff
@@ -303,22 +349,6 @@ function ExpandAggregateRow(props: {
     });
   }
 
-  // count(column) drilldown
-  if (aggregation === 'count') {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx:307]: Wouldn't we want to keep this?

**Review by markstory — COMMENTED** [review]
With the removal of the aggregate details view we could remove a bunch of code related to it. Things like pagination, the chart, drilldown condition generation in the tags table. Are you thinking of doing that cleanup work separately?

File: src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx:307]: @markstory  I think it was confirmed by Dora and Bowen that we remove this in light of the new navigation flow. 

In a follow up PR, I'll be adding the per-cell actions for `count(column)`; which was disabled temporarily until the navigation flow was implemented.


/cc  @doralchan @bowencai8

File: src/sentry/static/sentry/app/views/eventsV2/table/index.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/table/index.tsx:93]: Interesting. I'll revert this hunk.

Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/table/index.tsx]: Reverted in https://github.com/getsentry/sentry/pull/19332/commits/49d520e3beef265cb5a4438f619f57fd4955cbc0

**Review by doralchan — APPROVED** [review]
It looks like it works (says the designer)

**Pull request merged**